10_datastructures_lab/02.py

```diff
@@ -1,5 +1,35 @@
-"""
-Write a program that prints only unique values
-from sys.argv (without program name)
-"""
-
+"""
+Write a program that finds anagrams in a words file
+Program takes a path to a words file, 
+reads the file and searches for words with the same letters.
+All words with the same letters should be printed together
+"""
+
+_author_ = 'mariag'
+
+import sys
+from collections import Counter
+
+file_words = 'words.txt'
+wordsDic = {}
+
+#load key=word|value=letter counter to dictionary
+with open(file_words, 'r') as f_words:
+    for word in f_words:
+        wordsDic[word.rstrip('\n')] = Counter(word.rstrip('\n'))
+
+#print wordsDic
+delList =[]
+while len(wordsDic)>0:
+    for i in delList:
+        del wordsDic[i]
+    delList = []
+    print("")
+    word = wordsDic.keys()[0]
+    counter_i = wordsDic.get(word)
+    sys.stdout.write(word+" ")
+    del wordsDic[word]
+    for word_i in wordsDic:
+        if  counter_i == wordsDic.get(word_i):
+            sys.stdout.write(word_i+" ")
+            delList.append(word_i)
```

Comment:
This seems to be a bit too complex and as I can see it runs O(n^2)
You could make it simpler and faster by using a dictionary where the keys are the sorted version of each string and the values are the original strings. Each key that has more than one entry at the end has a list of anagrams.